File: src/connectfour/ui/Menu.java
package connectfour.ui;

import connectfour.controller.GameController;

public abstract class Menu {

	public abstract void saveGame();

	public abstract void loadSaveGame();

	public void newGame() {
		GameController.getInstance().newGame();
	}

	public abstract void quitGame();

	public abstract void show();

	public void exitGame() {
		System.exit(0);
	}

}

File: src/main/scala/connectfour/ui/Menu.scala
package connectfour.ui

import connectfour.controller.Connect4GameController


/**
 * Created by maharr on 19.10.15.
 */
abstract class Menu(controller: Connect4GameController) {

  def saveGame

  def loadSaveGame

  def newGame = {
    // TODO
    // controller.newGame
  }

  //TODO: Braucht an die Methode auch mit Spieler?
  //def newGameWithPlayer = controller.newGame(player1, player2)

  def quitGame

  def show

  def exitGame = System.exit(0)
}
